lib/userdb.c

```diff
@@ -57,8 +60,11 @@ resolve_group(const char *group, gint *gid)
   struct group *gr;
   gchar *endptr;
 
+  if (!gid)
+    return FALSE;
   *gid = 0;
-  if (!(*group))
+
+  if (!group || !(*group))
```

Comment:
The !group part should be assert. The !(*group) return FALSE part is ok.

Reply:
I restored the original form of userdb.c with no checks for NULL pointers.